tests/test_main.py

```diff
@@ -813,12 +824,22 @@ def check_filename(self, filename, check_names):
                 decorators[(node.lineno, node.name)] = []
 
         try:
-            code = compile(source.tree, source.filename, "exec")
+            code = compile(source.tree, source.filename, "exec", dont_inherit=True)
         except SyntaxError:
             # for example:
             # SyntaxError: 'return' outside function
             print("skip %s" % filename)
             return
+        except RecursionError:
```

Comment:
How does this happen?

Reply:
I don't know exactly. I remember cases where RecursionErrors where thrown for code like `"a" + "a" + "a" ...` or something like this. But I don't remember if it was in compile/parse.

I found no example to verify this, but I would keep the `except`